compiler/rustc_middle/src/ty/context.rs

```diff
@@ -2569,9 +2569,14 @@ impl<'tcx> TyCtxt<'tcx> {
         self.mk_ty_infer(TyVar(v))
     }
 
+    #[inline]
+    pub fn mk_const(self, kind: ty::ConstKind<'tcx>, ty: Ty<'tcx>) -> Const<'tcx> {
```

Comment:
Follow up idea:

Use `kind: impl Into<ty::ConstKind<'tcx>>` and remove mk_const_var, mk_const_param and mk_const_infer by adding the approriate From impls. Could also add one for ty::UnevaluatedConst, PlaceholderConst and ValTree and adjust the call sites sites to rely on the new impl